tests/jballerina-unit-test/src/test/resources/test-src/klass/service_type_resource_method_decl_neg.bal

```diff
@@ -23,3 +23,41 @@ type SType service object {
 service class SClass {
     *SType;
 }
+
+service class ROne {
+    resource function get foo() returns string {
+        return "foo";
+    }
+}
+
+// Not having a implementation for "get foo" resource method is expected as resource method is not part of the obj type.
+service class RTwo {
+    *ROne;
+}
+
+type RType service object {
+    *ROne;
+    *RTwo;
+};
+
+service class RTypeImpl {
+    *RType;
+}
+
+
+service class DoDone {
+    resource function 'do f() returns int => 0;
+
+    resource function done f() returns int => 0;
+
+    public function 'do() returns int => 0;
+}
+
+service class Do {
```

Comment:
Shall we add separate tests for the fix done for https://github.com/ballerina-platform/ballerina-lang/issues/31906 with mismatched function counts? May also have to add a runtime test if we don't have one already, right (IIRC it worked as expected)?

Reply:
You want me to move negative test's bal code to a separate file or to a separate test class?
The runtime typechecking approach is drastically different from the compiler's short circuiting approach

Reply:
No no, separate tests for cases like (without inclusion)

```ballerina
service class Foo {
    resource function get greeting() returns string => "hello";

    remote function hello() {
        
    }
}

service class Bar {
    remote function hello() {
        
    }
}

function positive() {
    Foo f = new Bar(); // allowed now
}

service class Baz {

}

function negative() {
    Foo f = new Baz(); // still not allowed
}
```

Re: runtime tests, just realized the runtime methods are modeled separately. Came across https://github.com/ballerina-platform/ballerina-lang/issues/31942 though. I guess we can add runtime tests when fixing that.

Reply:
fixed in https://github.com/ballerina-platform/ballerina-lang/pull/31761/commits/01c4a1d16ab1975e72e4f587af89037fa089d351